src/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/FunctionPointerOps.cs

```diff
@@ -90,7 +90,11 @@ public static unsafe IntPtr GetGenericMethodFunctionPointer(IntPtr canonFunction
 
                         // New generic descriptors are allocated on the native heap and not tracked in the GC.
                         UIntPtr allocationSize = new UIntPtr((uint)(c_genericDictionaryChunkSize * sizeof(RuntimeGeneratedGenericMethodDescriptor)));
+#if !PLATFORM_UNIX
                         IntPtr pNewMem = Interop.mincore.HeapAlloc(Interop.mincore.GetProcessHeap(), 0, allocationSize);
```

Comment:
We should factor this out into a common method to avoid these ifdefs. E.g. I would make it src/Common/src/Interop/MemoryHeap.cs